src/qibolab/sequence.py

```diff
@@ -69,7 +82,16 @@ def channel(self, channel: ChannelId) -> Iterable[PulseLike]:
 
     def channel_duration(self, channel: ChannelId) -> float:
         """Duration of the given channel."""
-        return sum(pulse.duration for pulse in self.channel(channel))
+        sequence = self
+        for _, pulse in self:
+            if isinstance(pulse, Align):
+                sequence = self.align_to_delays()
+                break
```

Comment:
Thanks, done in 4e4c2a5253efe823e5e1f7f1074eb316e164939d. Went with `any` since it is not that much of an overhead and as you say the alternative, despite simpler, is in principle suboptimal in most cases, since I do not expect `Align` to be used that much.